### ISSUE TITLE
Zero-length literal as a boolean

### ISSUE THREAD

**Issue by schwarja** [openlink/virtuoso-opensource#308]
Query:

```
SELECT ?X WHERE 
{ 
BIND ("" as ?X)
FILTER(?X)
}
```

Result is "", but according to SPARQL 1.1 documentation (http://www.w3.org/TR/sparql11-query/#ebv):

If the argument is a plain literal or a typed literal with a datatype of xsd:string, the EBV is false if the operand value has zero length; otherwise the EBV is true.

**Comment by HughWilliams** [comment]
@schwarja: what are you expecting this query to return as other triples stores like jena return no results  also ?

SQL> SPARQL SELECT ?X WHERE  {  BIND ("" as ?X) FILTER(?X) };
X
VARCHAR NOT NULL

---

1 Rows. -- 4 msec.
SQL>

**Comment by schwarja** [comment]
Exactly, I would expect empty result because BIND("" AS ?X) has zero length, it should be evaluated as false, therefore the condition FILTER(?X) is not satisfied. But Virtuoso returns one row with "" literal.

**Comment by HughWilliams** [comment]
@schwarja:  OK, this has been reported to development to look into ...
